### PR TITLE
add intel goldmont plus

### PR DESCRIPTION
**Pull request by ajax16384**

### CODE REVIEW

**Review by toor1245 — APPROVED** [review]

File: test/cpuinfo_x86_test.cc [hunk 1]
```diff
@@ -1138,6 +1138,21 @@ TEST_F(CpuidX86Test, INTEL_LAKEMONT) {
...
 
+// https://github.com/InstLatx64/InstLatx64/blob/master/GenuineIntel/GenuineIntel00706A8_GoldmontPlus_CPUID.txt
+TEST_F(CpuidX86Test, INTEL_GOLDMONT_PLUS) {
+  cpu().SetLeaves({
+      {{0x00000000, 0}, Leaf{0x00000018, 0x756E6547, 0x6c65746E, 0x49656E69}},
+      {{0x00000001, 0}, Leaf{0x000706A8, 0x00400800, 0x4FF8EBBF, 0xBFEBFBFF}},
+  });
+  const auto info = GetX86Info();
+
+  EXPECT_STREQ(info.vendor, "GenuineIntel");
```
Comment (toor1245) [inline test/cpuinfo_x86_test.cc:1149]: could you change `"GenuineIntel"` to `CPU_FEATURES_VENDOR_GENUINE_INTEL`, please?

Comment (ajax16384) [reply test/cpuinfo_x86_test.cc]: ok

File: include/cpuinfo_x86.h [hunk 2]
```diff
@@ -140,6 +140,7 @@ typedef enum {
   INTEL_BDW,           // BROADWELL
   INTEL_SKL,           // SKYLAKE
   INTEL_ATOM_GMT,      // GOLDMONT
+  INTEL_ATOM_GMT_P,    // GOLDMONT+
```
Comment (Mizux) [inline include/cpuinfo_x86.h:143]: line 172 below we've used `AMD_ZEN_PLUS` so I'm thinking if we should use `INTEL_ATOM_GMT_PLUS` instead :thinking:

Comment (ajax16384) [reply include/cpuinfo_x86.h]: ok

**Review by toor1245 — APPROVED** [review]

**Pull request merged**